core/system/src/mpu/vmpu.c

```diff
@@ -165,12 +165,79 @@ static void vmpu_sanity_check_box_namespace(int box_id, const char *const box_na
     } while (box_namespace[length]);
 }
 
+static void vmpu_box_index_init(uint8_t box_id, const uint32_t index_size, const UvisorBoxConfig * const config)
+{
+    void * box_bss;
+    UvisorBoxIndex * index;
+    uint32_t heap_size = config->heap_size;
+
+    if (box_id == 0)
+    {
+        if (!__uvisor_config.heap_start || !vmpu_sram_addr((uint32_t) __uvisor_config.heap_start)) {
+            HALT_ERROR(SANITY_CHECK_FAILED,
+                "Heap start pointer (0x%08x) is not in sram memory.\n",
+                (uint32_t) __uvisor_config.heap_start);
+        }
+        if (!__uvisor_config.heap_end || !vmpu_sram_addr((uint32_t) __uvisor_config.heap_end)) {
+            HALT_ERROR(SANITY_CHECK_FAILED,
+                "Heap end pointer (0x%08x) is not in sram memory.\n",
+                (uint32_t) __uvisor_config.heap_end);
+        }
+        if (__uvisor_config.heap_end < __uvisor_config.heap_start) {
+            HALT_ERROR(SANITY_CHECK_FAILED,
+                "Heap end pointer (0x%08x) is smaller than heap start pointer (0x%08x).\n",
+                (uint32_t) __uvisor_config.heap_end,
+                (uint32_t) __uvisor_config.heap_start);
+        }
+        /* Box 0 still uses the main heap to be backwards compatible. */
+        g_context_current_states[0].context = (uint32_t) __uvisor_config.heap_start;
```

Comment:
If this line is executed before calling `vmpu_acl_stack ()` (and in particular before [this line](https://github.com/ARMmbed/uvisor/pull/230/commits/c7d2a83117b7f080492e21a9c00b2eec2ad41e5a#diff-d5e475afa26cdc172546bc52c5d788abR572)) the context for box 0 is cleared (`NULL`).

Shall we move this line directly to that function?